lib/Dav/Faces/FaceRoot.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace OCA\Recognize\Dav\Faces;
+
+use OC\Metadata\IMetadataManager;
+use OCA\Recognize\Db\FaceCluster;
+use OCA\Recognize\Db\FaceClusterMapper;
+use OCA\Recognize\Db\FaceDetection;
+use OCA\Recognize\Db\FaceDetectionMapper;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\Files\IRootFolder;
+use OCP\IPreview;
+use OCP\ITagManager;
+use OCP\IUser;
+use Sabre\DAV\Exception\Forbidden;
+use Sabre\DAV\Exception\NotFound;
+use Sabre\DAV\ICollection;
+use Sabre\DAV\IMoveTarget;
+use Sabre\DAV\INode;
+
+class FaceRoot implements ICollection, IMoveTarget {
+	private FaceClusterMapper $clusterMapper;
+	private FaceCluster $cluster;
+	private IUser $user;
+	private FaceDetectionMapper $detectionMapper;
+	private IRootFolder $rootFolder;
+	private IMetadataManager $metadataManager;
+	private ITagManager $tagManager;
+	private IPreview $previewManager;
+	/**
+	 * @var \OCA\Recognize\Dav\Faces\FacePhoto[]
+	 */
+	private array $children = [];
+
+	public function __construct(FaceClusterMapper $clusterMapper, FaceCluster $cluster, IUser $user, FaceDetectionMapper $detectionMapper, IRootFolder $rootFolder, IMetadataManager $metadataManager, ITagManager $tagManager, IPreview $previewManager) {
+		$this->clusterMapper = $clusterMapper;
+		$this->cluster = $cluster;
+		$this->user = $user;
+		$this->detectionMapper = $detectionMapper;
+		$this->rootFolder = $rootFolder;
+		$this->metadataManager = $metadataManager;
+		$this->tagManager = $tagManager;
+		$this->previewManager = $previewManager;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getName() {
+		return $this->cluster->getTitle() !== '' ? $this->cluster->getTitle() : ''.$this->cluster->getId();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function setName($name) {
+		$this->cluster->setTitle(basename($name));
+		$this->clusterMapper->update($this->cluster);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function createDirectory($name) {
+		throw new Forbidden('Not allowed to create directories in this folder');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function createFile($name, $data = null) {
+		throw new Forbidden('Not allowed to create files in this folder');
+	}
+
+	/**
+	 * @throws \OCP\Files\NotPermittedException
+	 * @throws \OC\User\NoUserException
+	 */
+	public function getChildren(): array {
+		if (count($this->children) === 0) {
+			$this->children = array_map(function (FaceDetection $detection) {
+				return new FacePhoto($this->detectionMapper, $this->cluster, $detection, $this->rootFolder->getUserFolder($this->user->getUID()), $this->tagManager, $this->metadataManager, $this->previewManager);
+			}, $this->detectionMapper->findByClusterId($this->cluster->getId()));
+		}
+		return $this->children;
+	}
+
+	public function getChild($name): FacePhoto {
+		[$fileId,] = explode('-', $name);
+		try {
+			$detection = $this->detectionMapper->findByFileIdAndClusterId((int)$fileId, $this->cluster->getId());
```

Comment:
Maybe we can still use this->children as a cache here so it would not query if getChildren() has been called before

Reply:
Ah, nice. That's smart